### DISCUSSION TITLE
updateCheckRun mutation fails arbitrarily

### DISCUSSION THREAD

**Discussion by sanketsaurav** [community/community#24486] [Apps, API and Webhooks]
<p>The updateCheckRun mutation fails arbitrarily on some checks and returns null in the result.</p>
<pre><code>query = """
        mutation($input:UpdateCheckRunInput!) {
            updateCheckRun(input:$input) {
                checkRun {
                    databaseId
                    conclusion
                    status
                }
            }
        }
        """
</code></pre>
<p>for checkRunId ‘MDg6Q2hlY2tSdW40NDQxNjI4OA==’.</p>
<p>This happens completely arbitrarily. The GitHub app credentials used to make this query has required permissions to the repo.</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/sanketsaurav">@sanketsaurav</a>,</p>
<p>To clarify, when you say that it fails arbitrarily, you mean that you’re using the same input each time and it only fails some of the time, correct? Is this a particularly big query?</p>
<p>Thanks!</p>

**Comment by sanketsaurav** [comment, marked as answer]
<p>Hey <a class="mention" href="/that-pat">@that-pat</a>,</p>
<p>I was able to figure this out – this was failing for queries which have particularly high number of annotations. Apparently, the number of annotations <a href="https://developer.github.com/v3/checks/runs/#output-object-1" rel="nofollow noopener">must not be more than 50</a> in one query. The GraphQL API fails silently in such case without throwing any errors, and just returns <code>null</code>, as I mentioned. The REST API however shows the error properly, which is how I was able to debug this.</p>
<p>It would be great it the <a href="https://developer.github.com/v4/input_object/checkrunoutput/" rel="nofollow noopener">v4 documentation</a> mentions this as well.</p>
